### PR TITLE
renderdoc: init at version 0.34pre

### PR DESCRIPTION
**Pull request by jansol**
Initialising a few commits after the latest release due to some upstream
improvements to the build system.

###### Motivation for this change

It was missing. It's also a great app needed by multiple people, including me.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Ralith** [comment]
Thanks for putting this together! One question: Is vulkan-loader needed for renderdoc to debug vulkan use? If so, presumably it needs to actually be referenced somehow in the package (added to the rpath after unused paths are deleted?). If not, it should be removed from the package's arguments.

**Comment by Ralith** [comment]
After discussion in #renderdoc it sounds like vulkan-loader needs to go in the binary's rpath (in which case you'll have to be careful it doesn't get stripped as unused) or be added by the wrapper to `LD_LIBRARY_PATH`.

File: pkgs/applications/graphics/renderdoc/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  src = fetchFromGitHub {
+    owner = "baldurk";
+    repo = "renderdoc";
+
+    rev = "5e2717daec53e5b51517d3231fb6120bebbe6b7a";
+    sha256 = "1zpvjvsj5c441kyjpmd2d2r0ykb190rbq474nkmp1jk72cggnpq0";
+  };
+
+  buildInputs = [ cmake qtbase git pkgconfig xorg.libpthreadstubs xorg.libXdmcp qtx11extras ];
```
Comment (ndowens) [inline pkgs/applications/graphics/renderdoc/default.nix]: cmake, pkgconfig would go into nativeBuildInputs. Git should not be specified as a dependency

**Review by ndowens — CHANGES_REQUESTED** [review]

**Comment by jansol** [comment]
I'm not 100% sure if those patchelf invocations are the right thing to do, but it appears to work. (as in, creating and opening vulkan captures with it succeeds, whereas it fails without them)

**Comment by Ralith** [comment]
That seems like a reasonable solution, though I may be wrong.

Would it make sense to remove the LD_LIBRARY_PATH munging and do the same rpath magic for librenderdoc? It's a minor quibble, as either works, but consistency is nice.

**Comment by Ralith** [comment]
Actually, is DT_NEEDED meant to be used with absolute paths? My intuition says you're just supposed to have filenames there and let the rpath sort out the location.

File: pkgs/applications/graphics/renderdoc/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  name = "renderdoc-${version}";
+  version = "0.34pre";
+
+  src = fetchFromGitHub {
+    owner = "baldurk";
+    repo = "renderdoc";
+
```
Comment (ndowens) [inline pkgs/applications/graphics/renderdoc/default.nix]: Get rid of the empty space between repo and rev

File: pkgs/applications/graphics/renderdoc/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    wrapQtProgram $out/bin/qrenderdoc --suffix LD_LIBRARY_PATH : $out/lib --suffix LD_LIBRARY_PATH : ${vulkan-loader}/lib
+    mv $out/bin/renderdoccmd $out/bin/.bin/renderdoccmd
+    ln -s $out/bin/.bin/renderdoccmd $out/bin/renderdoccmd
+    wrapProgram $out/bin/renderdoccmd --suffix LD_LIBRARY_PATH : $out/lib --suffix LD_LIBRARY_PATH : ${vulkan-loader}/lib
+  '';
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "A single-frame graphics debugger";
+    homepage = "https://renderdoc.org/";
```
Comment (ndowens) [inline pkgs/applications/graphics/renderdoc/default.nix]: Can remove the quotes around url

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**